src/number_format.js

```diff
@@ -603,13 +611,13 @@ class NumberFormat extends React.Component {
    **/
   correctInputValue(caretPos: number, lastValue: string, value: string) {
     const {format, allowNegative, prefix, suffix} = this.props;
-    const {decimalSeparator} = this.getSeparators();
+    const {allowedDecimalSeparators, decimalSeparator} = this.getSeparators();
     const lastNumStr = this.state.numAsString || '';
     const {selectionStart, selectionEnd} = this.selectionBeforeInput;
     const {start, end} = findChangedIndex(lastValue, value);
 
-    /** Check if only . is added in the numeric format and replace it with decimal separator */
-    if (!format && decimalSeparator !== '.' && start === end && value[selectionStart] === '.') {
+    /** Check for any allowed decimal separator is added in the numeric format and replace it with decimal separator */
+    if (!format && start === end && allowedDecimalSeparators.includes(value[selectionStart])) {
```

Comment:
Note: This might not work in all browsers. Should I extract a util instead?

Reply:
You can use .indexOf instead.